src/components/FlyoutMenu/FlyoutMenu.test.jsx

```diff
@@ -0,0 +1,251 @@
+
+    const cancelButtonTest = await waitForElement(() => queryByText(cancelButtonText));
+
+    expect(applyButtonTest).toBeTruthy();
+    expect(cancelButtonTest).toBeTruthy();
+  });
+
+  it('Renders an open menu', async () => {
+    const { queryByText } = render(<FlyoutMenu defaultOpen triggerId="flyout-test" />);
+
+    const applyButtonTest = queryByText(applyButtonText);
+
+    const cancelButtonTest = queryByText(cancelButtonText);
+
+    expect(applyButtonTest).toBeNull();
+    expect(cancelButtonTest).toBeNull();
+  });
+
+  it('Renders closed', async () => {
+    const { queryByText } = render(<FlyoutMenu triggerId="flyout-test" />);
+
+    const applyButtonTest = await waitFor(() => queryByText(applyButtonText));
+
+    const cancelButtonTest = await waitFor(() => queryByText(cancelButtonText));
+
+    expect(applyButtonTest).toBeNull();
+    expect(cancelButtonTest).toBeNull();
+  });
+
+  it('Cancel Action', async () => {
+    const cancelAction = jest.fn();
+
+    const { queryByText } = render(
+      <FlyoutMenu open triggerId="flyout-test" transactional onCancel={cancelAction} />
+    );
+
+    const cancelButton = await waitForElement(() => queryByText(cancelButtonText));
+
+    fireEvent.click(cancelButton);
+
+    expect(cancelAction).toHaveBeenCalled();
+  });
+
+  it('Apply Action', async () => {
+    const applyAction = jest.fn();
+
+    const { queryByText } = render(
+      <FlyoutMenu open triggerId="flyout-test" transactional onApply={applyAction} />
+    );
+
+    const applyButton = await waitForElement(() => queryByText(applyButtonText));
+
+    fireEvent.click(applyButton);
+
+    expect(applyAction).toHaveBeenCalled();
+  });
+
+  it('Flyout closes when cancel triggered', async () => {
+    const { queryByText } = render(
+      <FlyoutMenu
+        triggerId="flyout-test"
+        defaultOpen
+        transactional
+        onApply={jest.fn()}
+        onCancel={jest.fn()}
+      />
+    );
+
+    const cancelButton = await waitForElement(() => queryByText(cancelButtonText));
+
+    fireEvent.click(cancelButton);
+
+    const applyButtonVisible = queryByText(applyButtonText);
+
+    expect(applyButtonVisible).toBeNull();
+  });
+
+  it('Flyout closes when apply triggered', async () => {
+    const { queryByText } = render(
+      <FlyoutMenu
+        triggerId="flyout-test"
+        defaultOpen
+        transactional
+        onApply={jest.fn()}
+        onCancel={jest.fn()}
+      />
+    );
+
+    const applyButton = await waitForElement(() => queryByText(applyButtonText));
+
+    fireEvent.click(applyButton);
+
+    const applyButtonVisible = queryByText(applyButtonText);
+
+    expect(applyButtonVisible).toBeNull();
+  });
+
+  it('Click to open and close menu', async () => {
+    const { container, queryByText } = render(<FlyoutMenu transactional triggerId="flyout-test" />);
+
+    const button = await container.firstChild.firstChild;
```

Comment:
I think you should query using the label (aria-label) instead of this container.firstChild.firstChild in order to better mimic what a human tester would do